WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m

```diff
@@ -667,15 +785,28 @@ - (NSManagedObjectContext *)managedObjectContext
 - (NSFetchRequest *)fetchRequest
 {
     NSString *sortKey               = NSStringFromSelector(@selector(timestamp));
-    NSArray *filteredNoteObjectIDs  = self.notificationIdsBeingDeleted.allObjects;
-    NSPredicate *predicate          = [NSPredicate predicateWithFormat:@"NOT (SELF IN %@)", filteredNoteObjectIDs];
     NSFetchRequest *fetchRequest    = [NSFetchRequest fetchRequestWithEntityName:self.entityName];
     fetchRequest.sortDescriptors    = @[[NSSortDescriptor sortDescriptorWithKey:sortKey ascending:NO] ];
-    fetchRequest.predicate          = predicate;
+    fetchRequest.predicate          = [self predicateForSelectedFilters];
     
     return fetchRequest;
 }
 
+- (NSPredicate *)predicateForSelectedFilters
+{
+    NSDictionary *filtersMap = @{
+        @(NotificationFilterUnread)     : @" AND (read = NO)",
+        @(NotificationFilterComment)    : @" AND (type = 'comment')",
+        @(NotificationFilterFollow)     : @" AND (type = 'follow')",
+        @(NotificationFilterLike)       : @" AND (type = 'like')"
```

Comment:
Might be nice to make `NoteTypeLike`, `NoteTypeFollow`, and `NoteTypeComment` in `Notification.m` external/public and reference the constants here.